data/src/main/java/com/linkedin/data/schema/compatibility/AnnotationCompatibilityChecker.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.data.schema.compatibility;
+
+import com.linkedin.data.schema.DataSchema;
+import com.linkedin.data.schema.NamedDataSchema;
+import com.linkedin.data.schema.PathSpec;
+import com.linkedin.data.schema.annotation.DataSchemaRichContextTraverser;
+import com.linkedin.data.schema.annotation.AnnotationCheckResolvedPropertiesVisitor;
+import com.linkedin.data.schema.annotation.SchemaAnnotationHandler;
+import com.linkedin.data.schema.annotation.SchemaAnnotationHandler.AnnotationCompatibilityResult;
+import com.linkedin.data.schema.annotation.SchemaAnnotationProcessor;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ *  Schema Annotation compatibility checker
+ *  The annotation compatibility check is part of the annotation processing framework.
+ *  If users use annotation processor to resolve properties,
+ *  they may also provide a annotation compatibility method to define the way to check annotation's compatibility.
+ *  In this checker, it will call the annotationCompatibilityCheck method which provide in the SchemaAnnotationHandler
+ *  by user to do the annotation compatibility check.
+ *
+ * @author Yingjie Bi
+ */
+public class AnnotationCompatibilityChecker
+{
+  private static final Logger _log = LoggerFactory.getLogger(AnnotationCompatibilityChecker.class);
+
+  /**
+   * Check the annotation compatibilities
+   * process olderSchema and newerSchema in the SchemaAnnotationProcessor to get the resolved result with resolvedProperties.
+   * then using the resolvedProperties to do the annotation compatibility check.
+   * @param olderSchema
+   * @param newerSchema
+   * @param handlers SchemaAnnotationHandler list
+   * @return List<AnnotationCompatibilityResult>
+   */
+  public static List<AnnotationCompatibilityResult> checkPegasusSchemaCompatibility(DataSchema olderSchema, DataSchema newerSchema,
```

Comment:
This function name seems confusing, in the comment it is meant for annotation check not the schema itself

Reply:
Please don't increase the length of `checkPegasusSchemaCompatibility` into `checkPegasusSchemaAnnoationaCompatibility`

Reply:
Change name to 'checkPegasusSchemaAnnotation'. Later on I will add new method 'checkExtensionSchemaAnnotation' for extension annotation check.